server/src/main/java/org/opensearch/tracer/Tracer.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracer;
+
+import java.util.Map;
+
+public interface Tracer {
+    /**
+     * Start the trace with passed attributes. It takes care of automatically propagating the context/parent to the child spans wherever the context switch is
+     * happening like from one thread to another in the same ExecutorService, across thread pools and even across nodes.
+     * <p>
+     * Returns a span reference which can be used for ending it.
+     */
+    default Span startTrace(String spanName, Map<String, String> attributes, Level level) {
```

Comment:
Are we expecting String only as attribute value?

Reply:
handled now